js/boaexchange.js

```diff
@@ -0,0 +1,1163 @@
+                'createMarketOrder': false,
+                'cancelOrder': true,
+                'cancelOrders': true,
+                'createDepositAddress': true,
+                'deposit': false,
+                'editOrder': 'emulated',
+                'fetchBalance': true,
+                'fetchBidsAsks': false,
+                'fetchClosedOrders': true,
+                'fetchCurrencies': true,
+                'fetchDeposits': true,
+                'fetchDepositAddress': true,
+                'fetchDepositAddresses': true,
+                'fetchFundingFees': true,
+                'fetchFundingLimits': true,
+                'fetchL2OrderBook': true,
+                'fetchLedger': true,
+                'fetchMarkets': true,
+                'fetchMyTrades': true,
+                'fetchOHLCV': true,
+                'fetchOpenOrders': true,
+                'fetchOrderBook': true,
+                'fetchOrderBooks': false,
+                'fetchOrder': true,
+                'fetchOrders': true,
+                'fetchTicker': true,
+                'fetchTickers': true,
+                'fetchTrades': true,
+                'fetchTradingFee': true,
+                'fetchTradingFees': true,
+                'fetchTradingLimits': true,
+                'fetchTransactions': true,
+                'fetchWithdrawals': true,
+                'transferToExchange': true,
+                'transfer': true,
+                'withdraw': true,
+            },
+            'hostname': 'boaexchange.com',
+            'timeframes': {
+                '1m': '1',
+                '5m': '5',
+                '15m,': '15',
+                '30m': '30',
+                '1h': '60',
+                '2h': '120',
+                '3h': '180',
+                '4h': '240',
+                '6h': '360',
+                '12h': '720',
+                '1d': '1440',
+                '1w': '10080',
+                '2w': '20160',
+                '1M': '43800',
+            },
+            'urls': {
+                'logo': 'https://boaexchange.com/4cdef72eb47d4a759d2c72e619f48827.png',
+                'api': 'https://api.{hostname}/api',
+                'www': 'https://boaexchange.com',
+                'doc': [
+                    'https://api.boaexchange.com/docs/',
+                ],
+            },
+            'api': {
+                'v1': {
+                    'get': [
+                        'addresses',
+                        'addresses/{label}',
+                        'airdrops',
+                        'balances',
+                        'balances/{label}',
+                        'chat',
+                        'coins',
+                        'coins/{label}',
+                        'coins/{label}/limits',
+                        'deposits',
+                        'deposits/{depositId}',
+                        'ledger',
+                        'markets',
+                        'markets/{label}',
```

Comment:
These all look like public endpoints to me, so we should not require credentials for any call, please, see how public/private APIs are structured, for example, here: https://github.com/ccxt/ccxt/blob/master/js/hitbtc2.js

So, we will need to work a bit more on the `sign()` and on `describe()`. I suggest that we structure it like:

```JavaScript
'api': {
    'public': {
        // everything that is public and doesn't require auth
        'get': [ ... ], 
        'post': [ ... ],
    },
    'private': {
        // everything that is private and requires auth
        'get': [ ... ], 
        'post': [ ... ],
        'delete': [ ... ],
    }
}
```

And rename the implicit calls accordingly, `this.v1Get...` → `this.publicGet...` or `this.privatePost..`

Reply:
One more thing, that also relates to other public methods as well, not just markets, but also, orderbooks and other public stuff.